runtime/drive_handler_test.go

```diff
@@ -0,0 +1,190 @@
+				PathOnHost: firecracker.String("/fake/stub/path2"),
+			},
+		},
+	}
+
+	expectedDriveIDs := []string{
+		"stub0",
+		"stub1",
+		"stub2",
+	}
+
+	for i, path := range expectedReplacements {
+		driveID, err := handler.PatchStubDrive(ctx, client, path)
+		assert.NoError(t, err, "failed to patch stub drive")
+		assert.Equal(t, expectedDriveIDs[i], firecracker.StringValue(driveID), "drive ids are not equal")
+	}
+}
+
+func TestPatchStubDrive_concurrency(t *testing.T) {
+	ctx := context.Background()
+	mockClient := &fctesting.MockClient{
+		PatchGuestDriveByIDFn: func(params *ops.PatchGuestDriveByIDParams) (*ops.PatchGuestDriveByIDNoContent, error) {
+			return nil, nil
+		},
+	}
+
+	fcClient := firecracker.NewClient("/path/to/socket", nil, false, firecracker.WithOpsClient(mockClient))
+	client, err := firecracker.NewMachine(ctx, firecracker.Config{}, firecracker.WithClient(fcClient))
+	assert.NoError(t, err, "failed to create new machine")
+
+	handler := stubDriveHandler{
+		drives: []models.Drive{
+			{
+				DriveID:    firecracker.String("stub0"),
+				PathOnHost: firecracker.String("/fake/stub/path0"),
+			},
+			{
+				DriveID:    firecracker.String("stub1"),
+				PathOnHost: firecracker.String("/fake/stub/path1"),
+			},
+			{
+				DriveID:    firecracker.String("stub2"),
+				PathOnHost: firecracker.String("/fake/stub/path2"),
+			},
+			{
+				DriveID:    firecracker.String("stub3"),
+				PathOnHost: firecracker.String("/fake/stub/path3"),
+			},
+			{
+				DriveID:    firecracker.String("stub4"),
+				PathOnHost: firecracker.String("/fake/stub/path4"),
+			},
+			{
+				DriveID:    firecracker.String("stub5"),
+				PathOnHost: firecracker.String("/fake/stub/path5"),
+			},
+			{
+				DriveID:    firecracker.String("stub6"),
+				PathOnHost: firecracker.String("/fake/stub/path6"),
+			},
+			{
+				DriveID:    firecracker.String("stub7"),
+				PathOnHost: firecracker.String("/fake/stub/path7"),
+			},
+			{
+				DriveID:    firecracker.String("stub8"),
+				PathOnHost: firecracker.String("/fake/stub/path8"),
+			},
+			{
+				DriveID:    firecracker.String("stub9"),
+				PathOnHost: firecracker.String("/fake/stub/path9"),
+			},
+			{
+				DriveID:    firecracker.String("stub10"),
+				PathOnHost: firecracker.String("/fake/stub/path10"),
+			},
+			{
+				DriveID:    firecracker.String("stub11"),
+				PathOnHost: firecracker.String("/fake/stub/path11"),
+			},
+		},
+	}
+
+	replacementPaths := []string{
+		"/correct/path0",
+		"/correct/path1",
+		"/correct/path2",
+		"/correct/path3",
+		"/correct/path4",
+		"/correct/path5",
+		"/correct/path6",
+		"/correct/path7",
+		"/correct/path8",
+		"/correct/path9",
+		"/correct/path10",
+		"/correct/path11",
+	}
+	var wg sync.WaitGroup
+	wg.Add(len(replacementPaths))
+	for _, path := range replacementPaths {
+		go func(path string) {
+			defer wg.Done()
+			_, err := handler.PatchStubDrive(ctx, client, path)
+			assert.NoError(t, err, "failed to patch stub drive")
+		}(path)
+	}
+
+	wg.Wait()
```

Comment:
After this `wg.Wait()`, can you assert that all of the paths are now equivalent to the `replacementPaths`?

Reply:
Done. Since this is done concurrently it doesn't guarantee that they will be in order, so have to use a map to ensure correctness.